src/pcnt.rs

```diff
@@ -0,0 +1,543 @@
+use core::fmt::Debug;
+
+
+use esp_idf_sys::esp;
+use esp_idf_sys::pcnt_config_t;
+use esp_idf_sys::pcnt_set_mode;
+use esp_idf_sys::pcnt_unit_t;
+use esp_idf_sys::EspError;
+
+use bitflags::bitflags;
+
+use crate::gpio::AnyInputPin;
+use crate::gpio::Pin;
+
+type UnitHandle = pcnt_unit_t;
+
+#[allow(dead_code)]
+#[derive(Debug, Copy, Clone)]
+pub enum PcntChannel {
+    Channel0,
+    Channel1,
+}
+
+impl Into<esp_idf_sys::pcnt_channel_t> for PcntChannel {
+    fn into(self) -> esp_idf_sys::pcnt_channel_t {
+        match self {
+            PcntChannel::Channel0 => esp_idf_sys::pcnt_channel_t_PCNT_CHANNEL_0,
+            PcntChannel::Channel1 => esp_idf_sys::pcnt_channel_t_PCNT_CHANNEL_1,
+        }
+    }
+}
+
+#[doc = " @brief PCNT channel action on signal edge"]
+#[derive(Debug, Copy, Clone)]
+pub enum PcntCountMode {
+    #[doc = "< Hold current count value"]
+    Hold,
+    #[doc = "< Increase count value"]
+    Increment,
+    #[doc = "< Decrease count value"]
+    Decrement,
+}
+
+impl Into<esp_idf_sys::pcnt_count_mode_t> for PcntCountMode {
```

Comment:
The convention in Rust is to implement `From`, not `Into`. They are symmetric anyway. In other words, you need to switch this guy (and all others below) to `impl From<PcntCountMode> for pcnt_count_mode_t`. Also, don't prefix with `esp_idf_sys::*` given that we are glob-importing from it.